src/EditorFeatures/CSharpTest/ConvertCast/ConvertTryCastToDirectCastTests.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.ConvertCast;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertConversionOperators
+{
+    using VerifyCS = CSharpCodeRefactoringVerifier<CSharpConvertTryCastToDirectCastCodeRefactoringProvider>;
+
+    [Trait(Traits.Feature, Traits.Features.ConvertCast)]
+    public class ConvertTryCastToDirectCastTests
+    {
+        [Fact]
+        public async Task ConvertFromAsToExplicit()
+        {
+            const string InitialMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = 1 as[||] object;
+    }
+}";
+            const string ExpectedMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = (object)1;
+    }
+}";
+            await new VerifyCS.Test
+            {
+                TestCode = InitialMarkup,
+                FixedCode = ExpectedMarkup,
+                CodeActionValidationMode = CodeActionValidationMode.Full,
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task ConvertFromAsToExplicit_ValueType()
+        {
+            const string InitialMarkup = @"
+class Program
+{
+    public static void Main()
+    {
+        var x = 1 as[||] byte;
+    }
+}";
+            await new VerifyCS.Test
+            {
+                TestCode = InitialMarkup,
```

Comment:
If `FixedCode` is not set, the refactoring provider is not called at all. The setup must to look like this 
```suggestion
                TestCode = InitialMarkup,
                FixedCode = InitialMarkup,
                OffersEmptyRefactoring = false,
```

Reply:
It's a bit sad, that this case is not covered any longer (The refactoring was behaving like a code fixer in such cases).